### PR TITLE
Make Vectorized ctor constexpr

### PR DESCRIPTION
**Pull request by swolchok**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #140047

constexpr vectorized seems nice to have.

Differential Revision: [D65623778](https://our.internmc.facebook.com/intern/diff/D65623778/)

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Comment by kimishpatel** [comment]
> constexpr vectorized seems nice to have.

Why? Just more modern way to code or something else?

File: aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp [hunk 1]
```diff
@@ -3104,8 +3104,6 @@ void quantized_groupnorm_nhwc_kernel(
     int64_t x_zp = X.q_zero_point();
     float x_scale = X.q_scale();
     fVec x_zp_vec((float)x_zp);
-    fVec one_vec(1.0f);
-    fVec zero_vec(0.0f);
```
Comment (malfet) [inline aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp:3108]: This looks unrelated

File: aten/src/ATen/cpu/vec/vec_base.h [hunk 2]
```diff
@@ -157,15 +157,15 @@ struct Vectorized {
...
+  constexpr Vectorized() : values{static_cast<T>(0)} {}
+  constexpr Vectorized(T val) {
     for (int i = 0; i != size(); i++) {
       values[i] = val;
     }
   }
   template<typename... Args,
            typename = std::enable_if_t<(sizeof...(Args) == size())>>
-  Vectorized(Args... vals) : values{vals...}{
+  constexpr Vectorized(Args... vals) : values{vals...}{
```
Comment (malfet) [inline aten/src/ATen/cpu/vec/vec_base.h:168]:
```suggestion
  constexpr Vectorized(Args... vals) : values{vals...} {
```

**Review by malfet — APPROVED** [review]

**Comment by swolchok** [comment]
Looks like we can't have nice things; the relevant intrinsics just aren't constexpr everywhere they need to be (e.g., https://godbolt.org/z/8z1a7z1Tz)
